strax/processing/hitlets.py

```diff
@@ -133,165 +136,90 @@ def refresh_hit_to_hitlets(hits, hitlets):
-            if temp_next_r_i == -1:
-                # There is no (more) previous record. So stop here and keep
-                # last next_r_i
-                break
-            next_r_i = temp_next_r_i
-            # There is a next record:
-            r = records[next_r_i]
-            data, (start, p_end_i) = _get_thing_data(hitlet, r)
-            if not start:
-                # If start is zero this means we have not found any
-                # overlap which should not have happened. In case of an
-                # overlap start and end should reflect the start and end
-                # sample of the hitlet for which we found data.
-                print('Data found for this record:', data,
-                      'Start index', start,
-                      'End index:', p_end_i)
-                raise ValueError('This is odd found the next record, but no'
-                                 ' overlapping indicies.')
-            temp_data[data_end:p_end_i] = data
-            data_end = p_end_i
-
-            if trial_counter > TRIAL_COUNTER_NEIGHBORING_RECORDS:
-                raise RuntimeError('Tried too hard. There are more than'
-                                   '100 successive records. This is odd...')
-            trial_counter += 1
-
-    # In some cases it might have happened that due to the left and right hit extension
-    # we extended our hitlet into regions without any data so we have to chop
-    # "time" according to the data we found....
-    time = hitlet['time'] + data_start * hitlet['dt']
-    temp_data = temp_data[data_start:data_end] + r['baseline'] % 1
-    return temp_data, time
-
-
-@numba.njit(nogil=True, cache=True)
-def _get_thing_data(thing, container):
-    """
-    Function which returns data for some overlapping indices of a thing
-    in a container. 
-    
-    Note:
-        Thing must be of the interval dtype kind.
-    """
-    overlap_hit_i, overlap_record_i = strax.overlap_indices(thing['time']//thing['dt'],
-                                                            thing['length'],
-                                                            container['time']//container['dt'],
-                                                            container['length'])
-    data = container['data'][overlap_record_i[0]:overlap_record_i[1]]
-    return data, overlap_hit_i
+    hitelts_is_single_row = isinstance(hitlets, np.void)
+    if hitelts_is_single_row:
+        # A structured array becomes void type if a single row is called,
+        # e.g. hitlets[0] which does not work in numba while, hitlets[:1]
+        # does. So we have to convert the row into the correct format first.
+        hitlets = np.array([hitlets])
+
+    data_field_in_hitlets = 'data' in hitlets.dtype.names
+    if data_field_in_hitlets:
+        data_is_not_empty = np.any(hitlets['data'] != 0)
+        if data_is_not_empty:
+            raise ValueError('The data field of hitlets must be empty!')
```

Comment:
Is this necessary? It is almost impossible someone would accidentally call this twice.

Reply:
Maybe you are right, I happened me during test. On the other hand this check does not cost much and adds another layer of security so I am more inclined to keep it.

Reply:
Okay, it's up to you. I was thinking about tpc data, where after saturation correction, one would like to update the peaklet waveform. But that won't happen here, so it doesn't matter.